lib/strategy/FragmentationStrategyProbabilityQuadHandling.ts

```diff
@@ -0,0 +1,84 @@
+import type * as RDF from '@rdfjs/types';
+import type { IQuadSink } from '../io/IQuadSink';
+import { FragmentationStrategyStreamAdapter } from './FragmentationStrategyStreamAdapter';
+import { FragmentationStrategySubject } from './FragmentationStrategySubject';
+
+export class FragmentationStrategyProbabilityQuadHandling extends FragmentationStrategyStreamAdapter {
+  private readonly generationProbability: number;
+  private readonly strategy: FragmentationStrategyStreamAdapter;
```

Comment:
I didn't use the `IFragmentationStrategy` because I think I would need to iterate the stream a first time to prune the quads and then pass it to the wrapped strategy.

Reply:
I don't see why we can't use `IFragmentationStrategy` tbh.
Using `FragmentationStrategyStreamAdapter` makes this fragmenter not as generic as it needs to be.